src/Common/src/Interop/Unix/libc/Interop.gethostname.cs

```diff
@@ -18,20 +19,20 @@ internal static unsafe string gethostname()
             const int ArrLength = HOST_NAME_MAX + 1;
 
             byte* name = stackalloc byte[ArrLength];
-            int errno = gethostname(name, ArrLength);
-            if (errno == 0)
+            int err = gethostname(name, ArrLength);
+            if (err != 0)
             {
-                int nullPos = 0;
-                for (; nullPos < ArrLength && name[nullPos] != '\0'; nullPos++) ;
-                return Encoding.UTF8.GetString(name, nullPos);
+                // This should never happen.  According to the man page,
+                // the only possible errno for gethostname is ENAMETOOLONG,
+                // which should only happen if the buffer we supply isn't big
+                // enough, and we're using a buffer size that the man page
+                // says is the max for POSIX (and larger than the max for Linux).
+                Debug.Fail("gethostname failed");
+                throw new InvalidOperationException();
```

Comment:
Perhaps capture the error number in the exception message? That way if this ever triggers in the wild we'll have a bit more info.